pkg/server/server_test.go

```diff
@@ -218,3 +222,72 @@ func TestAddRemoveFrontends(t *testing.T) {
 		t.Errorf("expected %v, got %v", e, a)
 	}
 }
+
+func TestServerProxyNormalClose(t *testing.T) {
```

Comment:
Before we have other tests sharing these definitions, let's first start in this way.
I am planning to add more UT with gomock to protect further behaviors and make it simpler for others contributors to write UT for their changes.

Reply:
This test coverage looks good. Would be nice to also add coverage of the DialResponse "with error" case that you fixed above.

Reply:
Sure, of course